Automattic/woocommerce-payments
# fix: test mode flag

Fixes https://github.com/Automattic/woocommerce-payments/issues/2450

#### Changes proposed in this Pull Request









The test mode flag is not being retrieved correctly, causing the store to be in a permanent "test mode" state after saving the settings.

#### Testing instructions





I tested this in my `wpsandbox.me` site (not locally), so that the "dev mode" check would be falsy.
You can also add `add_filter( 'wcpay_dev_mode', '__return_false' );` on your site to set dev mode to false.

- Go to the settings page
- Enable test mode
- Save the settings
- Refresh the page
- Notice the "test mode" notice being added
- Disable test mode
- Save the settings
- Refresh the page
- The "test mode" notice is removed
- Checkout allows test cards

Before: it wasn't allowing checkout with live card numbers
![Screen Shot 2021-07-21 at 10 31 05 AM](https://user-images.githubusercontent.com/273592/126518520-c25f7fb8-6c87-4b2b-8803-4ffb343542f9.png)

After: it allows checkout with live card numbers
![Screen Shot 2021-07-21 at 10 31 14 AM](https://user-images.githubusercontent.com/273592/126518595-691377bb-9472-4bb7-9b01-8c3568653cb6.png)



-------------------
